src/senaite/core/content/worksheettemplate.py

```diff
@@ -0,0 +1,570 @@
+    def setRestrictToMethod(self, value):
+        mutator = self.mutator("restrict_to_method")
+        mutator(self, value)
+
+    # BBB: AT schema field property
+    RestrictToMethod = property(getRestrictToMethod, setRestrictToMethod)
+
+    @security.protected(permissions.View)
+    def getRawInstrument(self):
+        instrument = self.getInstrument()
+        if instrument:
+            return instrument.UID()
+        return None
+
+    @security.protected(permissions.View)
+    def getInstrument(self):
+        accessor = self.accessor("instrument")
+        return accessor(self)
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setInstrument(self, value):
+        mutator = self.mutator("instrument")
+        mutator(self, value)
+
+    # BBB: AT schema field property
+    Instrument = property(getInstrument, setInstrument)
+
+    @security.protected(permissions.View)
+    def getEnableMultipleUseOfInstrument(self):
+        accessor = self.accessor("enable_multiple_use_of_instrument")
+        return accessor(self)
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setEnableMultipleUseOfInstrument(self, value):
+        mutator = self.mutator("enable_multiple_use_of_instrument")
+        mutator(self, value)
+
+    # BBB: AT schema field property
+    EnableMultipleUseOfInstrument = property(getEnableMultipleUseOfInstrument,
+                                             setEnableMultipleUseOfInstrument)
+
+    @security.protected(permissions.View)
+    def getNumOfPositions(self):
+        accessor = self.accessor("num_of_positions")
+        return accessor(self)
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setNumOfPositions(self, value):
+        mutator = self.mutator("num_of_positions")
+        mutator(self, value)
+        layout = self.getTemplateLayout()
+        len_layout = len(layout)
+        nums = int(value)
+        if nums == 0:
+            layout = []
+        elif len_layout == 0 and nums > 0:
+            layout = default_layout_positions(nums)
+        elif nums > len_layout:
+            rows = nums - len_layout
+            layout = layout + default_layout_positions(rows, len_layout)
+        elif nums < len_layout:
+            layout = layout[:nums]
+        self.setTemplateLayout(layout)
+
+    NumOfPositions = property(getNumOfPositions, setNumOfPositions)
+
+    @security.protected(permissions.View)
+    def getTemplateLayout(self):
+        accessor = self.accessor("template_layout")
+        return accessor(self) or []
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setTemplateLayout(self, value):
+        mutator = self.mutator("template_layout")
+        mutator(self, value)
+        num_pos = len(value)
+        if num_pos != self.getNumOfPositions():
+            self.setNumOfPositions(num_pos)
+
+    # BBB: AT schema field property
+    Layout = property(getTemplateLayout, setTemplateLayout)
```

Comment:
Hi Ramon!
Can not declare method `getLayout` because this method invoke from parent Plone view and trow exception:
```shell
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 176, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 385, in publish_module
  Module ZPublisher.WSGIPublisher, line 264, in publish
  Module ZPublisher.BaseRequest, line 450, in traverse
  Module plone.rest.patches, line 15, in __before_publishing_traverse__
  Module Products.CMFCore.DynamicType, line 164, in __before_publishing_traverse__
  Module Products.CMFDynamicViewFTI.fti, line 239, in queryMethodID
  Module Products.CMFDynamicViewFTI.fti, line 201, in defaultView
  Module Products.CMFPlone.PloneTool, line 696, in browserDefault
  Module bika.lims, line 195, in wrapper
```

[`Products.CMFPlone.PloneTool:695`](https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/PloneTool.py#L695)

`bika.lims, line 195` is deprecation decorator

_this example from my instance_

Reply:
That is why the `layout` field was renamed to `template_layout`: https://github.com/senaite/senaite.core/pull/2599#issuecomment-2376098725

Reply:
without annotation of deprecation:
```shell
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 176, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 385, in publish_module
  Module ZPublisher.WSGIPublisher, line 264, in publish
  Module ZPublisher.BaseRequest, line 450, in traverse
  Module plone.rest.patches, line 15, in __before_publishing_traverse__
  Module Products.CMFCore.DynamicType, line 164, in __before_publishing_traverse__
  Module Products.CMFDynamicViewFTI.fti, line 239, in queryMethodID
  Module Products.CMFDynamicViewFTI.fti, line 201, in defaultView
  Module Products.CMFPlone.PloneTool, line 696, in browserDefault
TypeError: getLayout() got an unexpected keyword argument 'check_exists'
```
...the same

Reply:
@ramonski pls, check that is above

Reply:
...

I think that no possibility use this method